src/shell/commands/disk_rebalance.cpp

```diff
@@ -0,0 +1,288 @@
+// Copyright (c) 2017, Xiaomi, Inc.  All rights reserved.
+// This source code is licensed under the Apache License Version 2.0, which
+// can be found in the LICENSE file in the root directory of this source tree.
+
+#include "shell/commands.h"
+#include "shell/argh.h"
+
+#include <math.h>
+#include <fmt/ostream.h>
+#include <dsn/utility/errors.h>
+#include <dsn/utility/output_utils.h>
+#include <dsn/utility/string_conv.h>
+#include <dsn/dist/replication/duplication_common.h>
+
+bool validate_cmd(const argh::parser &cmd,
+                  const std::set<std::string> &params,
+                  const std::set<std::string> &flags)
+{
+    if (cmd.size() > 1) {
+        fmt::print(stderr, "too many params!\n");
+        return false;
+    }
+
+    for (const auto &param : cmd.params()) {
+        if (params.find(param.first) == params.end()) {
+            fmt::print(stderr, "unknown param {} = {}\n", param.first, param.second);
+            return false;
+        }
+    }
+
+    for (const auto &flag : cmd.flags()) {
+        if (params.find(flag) != params.end()) {
+            fmt::print(stderr, "missing value of {}\n", flag);
+            return false;
+        }
+
+        if (flags.find(flag) == flags.end()) {
+            fmt::print(stderr, "unknown flag {}\n", flag);
+            return false;
+        }
+    }
+
+    return true;
+}
+
+std::unique_ptr<std::ostream> get_out_stream(const std::string &file_name)
+{
+    auto ostream_ptr = std::unique_ptr<std::ostream>();
+    if (file_name.empty()) {
+        ostream_ptr = std::unique_ptr<std::ostream>(&std::cout);
+    } else {
+        ostream_ptr = std::unique_ptr<std::ofstream>(new std::ofstream(file_name));
+        if (!*ostream_ptr) {
+            fmt::print(stderr, "open output file {} failed\n", file_name);
+            return nullptr;
+        }
+    }
```

Comment:
```suggestion
    std::unique_ptr<std::ostream> ostream_ptr;
    if (file_name.empty()) {
        ostream_ptr.reset(&std::cout);
    } else {
        ostream_ptr = make_unique<std::ofstream>(file_name);
        if (!*ostream_ptr) {
            fmt::print(stderr, "open output file {} failed\n", file_name);
            return nullptr;
        }
    }
```

Reply:
https://github.com/XiaoMi/pegasus/pull/498#discussion_r396491993 is better！